R/pretty_lollipop_plot.R

```diff
@@ -2,58 +2,118 @@
 #'
 #' @description Generates a visually appealing and interactive lollipop plot.
 #'
-#' @details This function is depending on a modified version of `readMAF` from the `g3viz` package.
-#' Returned plot is interactive, meaning the user can hover over individual points in the plot to reveal more information.
-#' The plot can also be exported in a variety of file formats inside the interactive view of the lollipop plot.
+#' @details Retrieve a maf of a specific sample or a set of samples. Returned plot is interactive, meaning the 
+#' user can hover over individual points in the plot to reveal more information. The plot can also be exported 
+#' in a variety of file formats inside the interactive view of the lollipop plot.
 #'
-#' @param maf_df A data frame containing the mutation data (from a MAF).
-#' @param gene The gene symbol to plot.
-#' @param plot_title Optional (defaults to gene name).
-#' @param plot_theme Options: cbioportal(default), blue, simple, nature, nature2, ggplot2, and dark.
-#' @param out_name Optional, set the file name of the plot, if you export it to disk. Default name is my_lollipop_plot_{gene}.
-#'
-#' @return Nothing.
+#' @param maf_data A data frame containing the mutation data (from a MAF).
+#' @param gene_symbol The gene symbol to plot.
+#' 
+#' @return A lolipop plot.
 #'
-#' @import g3viz dplyr
+#' @import tidyverse BiocManager maftools
 #' @export
 #'
 #' @examples
 #' library(GAMBLR.data)
 #' 
-#' #get metadata (Fl and DLBCL)
-#' metadata = get_gambl_metadata()
-#' this_meta = dplyr::filter(metadata, pairing_status == "matched")
-#' this_meta = dplyr::filter(this_meta, pathology %in% c("FL", "DLBCL"))
+#' #get metadata (BL_Thomas)
+#' metadata <- get_gambl_metadata() %>%
+#'     filter(cohort == "BL_Thomas")
 #' 
-#' maf = get_ssm_by_sample(these_samples_metadata = this_meta)
+#' maf_data <- get_ssm_by_samples(
+#'     these_samples_metadata = metadata
+#' )
 #'
 #' #construct pretty_lollipop_plot.
-#' pretty_lollipop_plot(maf_df = maf,
-#'                      gene = "MYC",
-#'                      plot_title = "Mutation data for MYC",
-#'                      plot_theme = "nature2")
+#' lolipop_result <- get_lolipop_by_gene_symbol(maf_data, "MYC")
+#' print(lolipop_result)
 #'
-pretty_lollipop_plot = function(maf_df,
-                                gene,
-                                plot_title,
-                                plot_theme = "cbioportal",
-                                out_name = paste0("my_lollipop_plot_", gene)){
-  if(missing(gene)){
-    stop("Plese provide a gene...")
+get_lolipop_by_gene_symbol <- function(maf_data, 
+                                       gene_symbol = NULL) {
+  # Read the MAF object if it's not already in MAF format
+  if (!inherits(maf_data, "MAF")) {
+    maf_data <- read.maf(maf_data)
   }
+  
+  maf_data <- maf_data@data 
+  # Filter for the specific gene 
+  gene_df <- maf_data %>% 
+    filter(Hugo_Symbol == gene_symbol) %>%
```

Comment:
This place can be a good example of inconsistent formatting - can we please maintain a tab indent instead of 2-space indent? This might have not been the case already but we should take this function overhaul to ensure we stick to the best practices